### PR TITLE
Hide symbols

### PR DESCRIPTION
**Pull request by daniellowell**
Reduces number of visible symbols.

### CODE REVIEW

File: miopengemm/CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,27 @@ file(GLOB_RECURSE source_files src/*.cpp)
...
+          };
+      local:
+          *;
+      };
+    ")
+
+    target_link_libraries(miopengemm PRIVATE "-Wl,--version-script=${CMAKE_CURRENT_BINARY_DIR}/lib.def")
+    target_link_libraries(miopengemm PRIVATE "-Wl,--exclude-libs,ALL")
+    set_target_properties(miopengemm PROPERTIES VISIBILITY_INLINES_HIDDEN 1)
+    target_link_libraries(miopengemm PUBLIC ${OPENCL_LIBRARIES} ${OpenBLAS_LIB} ${CLBLAST_LIB} ${ISAAC_LIB})
```
Comment (pfultz2) [inline miopengemm/CMakeLists.txt]: Why was this moved under the `if`?

Comment (daniellowell) [reply miopengemm/CMakeLists.txt]: Seemed logical. I can move it back.

Comment (daniellowell) [reply miopengemm/CMakeLists.txt]: Moved outside of if
